frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -151,45 +151,67 @@ export const GRAPH_DATA_SETTINGS = {
     dashboard: false,
     useRawSeries: true,
   },
+  "graph.series_order_dimension": {
+    getValue: (_series, settings) => settings["graph.dimensions"][1],
+    // This read dependency is set so that "graph.series_order" is computed *before* this value, ensuring that
+    // that it uses the stored value if one exists. This is needed to check if the dimension has actually changed
+    readDependencies: ["graph.series_order"],
+  },
   "graph.series_order": {
     section: t`Data`,
     widget: ChartSettingOrderedSimple,
     marginBottom: "1rem",
-    isValid: (series, settings) => {
+
+    getValue: (series, settings) => {
+      const seriesKeys = series.map(s => keyForSingleSeries(s));
```

Comment:
This could be a problem for row chart (and funnel chart too since the code is almost the same) as shown in the video.

Basically, `keyForSingleSeries(s)` would return `undefined`

https://www.loom.com/share/80045780c8c44d44996dd44fcf0e85b8